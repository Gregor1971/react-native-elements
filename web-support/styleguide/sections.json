[
  {
    "name": "Buttons",
    "components": "../../src/buttons/[A-Z]*.js"
  },
  {
    "name": "Cards",
    "components": "../../src/+(card|pricing)/[A-Z]*.js"
  },
  {
    "name": "Forms",
    "components": "../../src/form/[A-Z]*.js"
  },
  {
    "name": "Icons",
    "components": "../../src/+(icons|social)/[A-Z]*.js"
  },
  {
    "name": "Inputs",
    "components": "../../src/+(checkbox|rating|slider)/[A-Z]*.js"
  },
  {
    "name": "Lists",
    "components": "../../src/list/[A-Z]*.js"
  },
  {
    "name": "Tiles",
    "components": "../../src/tile/[A-Z]*.js"
  },
  {
    "name": "misc",
    "components":
      "../../src/+(avatar|badge|divider|header|text)/+(badge|[A-Z])*.js"
  }
]
