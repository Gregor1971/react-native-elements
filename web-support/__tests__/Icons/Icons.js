import parseSamples from 'enzyme-styleguidist-sample-parser';
import sections, { options } from '../../samples';

parseSamples({ Icons: sections.Icons }, options);
