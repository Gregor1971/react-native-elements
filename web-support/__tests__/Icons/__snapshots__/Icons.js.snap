// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons: Icon: [ prop: color ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="#071"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: component ] enzyme-shallow: snapshot 1`] = `
<TouchableWithoutFeedback
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</TouchableWithoutFeedback>
`;

exports[`Icons: Icon: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      Object {
        "padding": 50,
      },
    ]
  }
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: iconStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "transform": Array [
            Object {
              "rotateZ": "-45deg",
            },
          ],
        },
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: name ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="black"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: onLongPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onLongPress={[Function]}
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</TouchableHighlight>
`;

exports[`Icons: Icon: [ prop: onPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</TouchableHighlight>
`;

exports[`Icons: Icon: [ prop: raised ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      75,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      76,
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: reverse ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      75,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="white"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: reverseColor ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      75,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="#071"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: size ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "borderRadius": 54,
        "height": 104,
        "width": 104,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="black"
    name="pets"
    size={50}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: type ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <[object Object]
    color="black"
    name="sms"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Icons: Icon: [ prop: underlayColor ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  style={
    Array [
      false,
      Object {
        "borderRadius": 28,
        "height": 52,
        "width": 52,
      },
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="yellow"
>
  <[object Object]
    color="black"
    name="pets"
    size={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  />
</TouchableHighlight>
`;

exports[`Icons: SocialIcon: [ prop: button ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: component ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: disabled ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={true}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: fontFamily ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          Object {
            "fontFamily": "Courier New",
          },
          undefined,
          undefined,
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: fontStyle ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 50,
          },
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: fontWeight ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          Object {
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: iconColor ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="#071"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: iconSize ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={40}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: iconStyle ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "rotateZ": "-45deg",
              },
            ],
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: light ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      Object {
        "backgroundColor": "white",
      },
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="#02b875"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: loading ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      medium
    </TextElement>
    <ActivityIndicator
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="large"
      style={
        Array [
          101,
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": undefined,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: onLongPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
  underlayColor="black"
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;

exports[`Icons: SocialIcon: [ prop: onPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
  underlayColor="black"
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;

exports[`Icons: SocialIcon: [ prop: raised ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      false,
      95,
      false,
      false,
      Object {
        "borderRadius": 48,
        "height": 52,
        "width": 52,
      },
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: style ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      Object {
        "transform": Array [
          Object {
            "rotateZ": "-45deg",
          },
        ],
        "width": 120,
      },
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: title ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      96,
      false,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Array [
          99,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: type ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: [ prop: underlayColor ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Array [
      97,
      95,
      false,
      100,
      false,
      Object {
        "backgroundColor": "#02b875",
      },
      undefined,
      undefined,
    ]
  }
  underlayColor="black"
>
  <View
    style={98}
  >
    <[object Object]
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;
